index.bs

```diff
@@ -4841,10 +4841,12 @@ partial interface MLGraphBuilder {
 
 <details open algorithm>
   <summary>
-    To <dfn for="MLGraphBuilder">calculate reduction output sizes</dfn>, given a [=/list=] of unsigned integers |inputShape|, a optional [=/list=] of unsigned integers |axes|, and [=/boolean=] |keepDimensions|, perform the following steps. They return a new [=/list=] of unsigned integers.
+    To <dfn for="MLGraphBuilder">calculate reduction output sizes</dfn>, given a [=/list=] of unsigned integers |inputShape|, a optional [=/list=] of unsigned integers |axes|, and [=/boolean=] |keepDimensions|, perform the following steps. They return a new [=/list=] of unsigned integers, or failure.
   </summary>
     1. Let |inputRank| be |inputShape|'s [=list/size=].
+    1. If |inputRank| is 0, then return failure.
```

Comment:
Nop reductions are valid (where `axes` are empty), and scalars are just another instance of a nop:

```
x = tensorflow.constant([3,3,3], dtype=tensorflow.float32)
y = tensorflow.reduce_sum(x, axis=[])

# value: tf.Tensor([3. 3. 3.], shape=(3,), dtype=float32)
# shape: (3,)

x = tensorflow.constant([], dtype=tensorflow.float32)
y = tensorflow.reduce_sum(x, axis=[])

# value: tf.Tensor([], shape=(0,), dtype=float32)
# shape: (0,)
```

What's *not* valid is a scalar reduction with *non-empty* axes, because that would index axes beyond the rank. That means this restriction on 4847 is both redundant (because the check that each axis < input rank would validate that anyway) and exclusionary. So, we should remove it and update Chromium accordingly.